src/System.Buffers/src/System/ByteSpan.cs

```diff
@@ -74,7 +75,8 @@ public int Length
 
         public ByteSpan Slice(int index)
         {
-            Precondition.Require(index < Length);
+            if (index >= Length)
+                return new ByteSpan(_data, 0);
```

Comment:
Correct but I have another example. Let's say your input is byte array with invalid utf8 character near the end. I.e. First byte of the codepoint has encoded it's width of 4 but it is the last character of the span, if I just try to advance the pointer it would be 0 which I can easily check, if you fail fast I can't do anything about it and have to check (currentpos + charlen >= arrlen)